sotodlib/io/imprinter.py

```diff
@@ -682,9 +686,132 @@ def delete_book_files(self, book):
             self.logger.warning(f"Failed to remove {book.path}: {e}")
             self.logger.error(traceback.format_exc())
 
+    def get_book_times(self, book):
+        """
+        Given a book of co-sampled data with a timing system, this function will
+        return a gap-filled list of timestamps that should be contained in the book.
```

Comment:
So this does the "trimming" for you ... maybe mention that.

This is an opportunity to report how much time has been trimmed from the various filesets to make them all the same length... but I imagine that's been checked carefully elsewhere.

Reply:
Right. where do you think would be a good place to report this? should it make its way into the M_index file?

Reply:
I don't think I've seen the trimming amount reported anywhere yet.

Reply:
Ok cool, I think I'm going to leave it for now and open an issue, since I think this should be done by the book-binder, which creates the M_index file